src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -110,8 +109,15 @@ async function genPublicEphemeralKey(curve, Q) {
       break;
     case 'node':
       return nodePublicEphemeralKey(curve, Q);
+    default: {
+      const nobleCurve = await util.getNobleCurve(enums.publicKey.ecdh, curve.name);
+      const { publicKey: V, privateKey: v } = await curve.genKeyPair();
+
+      // The output includes parity byte
+      const rawSharedSecret = nobleCurve.getSharedSecret(v, Q);
+      return { publicKey: V, sharedKey: rawSharedSecret.subarray(1) };
```

Comment:
I'd avoid "privateKey" and "publicKey" because they are confusing in a context where we also have long-term private and public keys (e.g. `Q`).
The letters are convenient to follow what's going on, e.g. `vQ === vdG === dvG === dV`, and they are commonly used (e.g. see X25519 RFC).
Since we already have params named `d` and `Q`, I think switching to different names here is confusing, as the function is not generic.
If we really want to rename, I'd go with `longTermKeyPair` vs `ephemeralKeyPair`. But in a separate commit, so that it's clearer how the values map to the old ones with the lib switch.

Reply:
OK, fair enough. Yeah, I'd prefer to have more descriptive names but indeed we can do it separately :+1: